Release Candidate 0.2.3

jklenzing opened the pull request:
# Description

Test Branch for the release candidate

# How Has This Been Tested?

Testing with runs for current projects

**Test Configuration**:
* Mac OS X 10.15.7
* Python 3.8.2

# Checklist:

- [na] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

jklenzing commented:
TODO: 

- [ ] regenerate netcdf files with new metadata
- [ ] check with pysatModels
- [ ] try to upload to testpypi

jklenzing commented:
Pypi compatibility is not required for this version, but I've uploaded a version to the test site to better document issues for the next version.  https://test.pypi.org/project/sami2py/0.2.3/

aburrell commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,68 @@
+        include:
+          - python-version: 3.7
+            numpy_ver: 1.17
+            os: ubuntu-latest
+          - python-version: 3.7
+            numpy_ver: 1.17
+            os: windows-latest
```
Is it necessary to require OS here if you want to run both?

jklenzing replied:
Yes.  Matrix gives the product of all options above, while include adds single case elements

aburrell commented on docs/introduction.rst:
```diff
@@ -6,28 +6,50 @@
-
-The implementation used here includes several added options to the original release of SAMI2.  A full list is included in :ref:`modifications`, but several of these include:
- - The ability to scale the neutral atmosphere in which the ions form through direct modification of the exospheric neutral temperature for extreme solar minimum conditions, as discussed by Emmert et al [2010].
+Sami2py is a python module that runs the SAMI2 model, as well as archives, loads
```
Is it really `Sami2py` and not `sami2py` or `SAMI2Py`?

jklenzing replied:
fixed

aburrell commented on docs/introduction.rst:
```diff
@@ -6,28 +6,50 @@
-When referring to this software package, please cite the original paper by Huba et al [2000] https://doi.org/10.1029/2000JA000035 as well as the package by Klenzing and Smith [2019] https://doi.org/10.5281/zenodo.2875800.
+When referring to this software package, please cite the original paper by Huba
+et al [2000] https://doi.org/10.1029/2000JA000035 as well as the package by
+Klenzing and Smith [2019] https://doi.org/10.5281/zenodo.2875800.
```
```suggestion
Klenzing et al. [2019] https://doi.org/10.5281/zenodo.2875800.
```

aburrell commented on docs/sample_workflow.rst:
```diff
@@ -19,17 +25,23 @@ sami2py will raise an error if this is not done before trying to run the model.
 
-The MetaData can be accessed directly via the dictionary, or through the __repr__.  Typing
+The MetaData can be accessed directly via the dictionary, or through the
+__repr__.  Typing
```
```suggestion
``__repr__``.  Typing
```

aburrell commented on sami2py/__init__.py:
```diff
@@ -20,7 +20,10 @@
 import os
 
-__version__ = str('0.2.0')
+# set the version
+here = os.path.abspath(os.path.dirname(__file__))
+with open(os.path.join(here, 'version.txt')) as version_file:
+    __version__ = version_file.read().strip()
 
 # get home directory
 home_dir = os.path.expanduser('~')
```
Recommend deleting variables you don't want users to have access to at the top level of the module

jklenzing replied:
cleaned.  Added #151 as a breaking change to further simplify storage of required paths

aburrell commented on sami2py/_core.py:
```diff
@@ -45,7 +49,7 @@ def run_model(tag='model_run', lat=0, lon=0, alt=300, year=2018, day=1,
               Tinf_scale=1, Tn_scale=1., euv_scale=1,
               wind_scale=1, hwm_model=14,
-              fejer=True, ExB_drifts=np.zeros((10, 2)), ve01=0., exb_scale=1,
+              fejer=True, ExB_drifts=None, ve01=0., exb_scale=1,
```
Style note: why `ExB_drifts` and `exb_scale`? Also `Tinf_scale` and `Tn_scale`, though these are less annoying.

jklenzing replied:
Documented in #150

aburrell commented on sami2py/_core.py:
```diff
@@ -381,8 +403,6 @@ def _archive_model(path, clean, fejer, fmtout, outn):
     void
         Moves files to the archive directory
```
Recommend removing Returns here.  Can move description to `Notes`

aburrell commented on sami2py/_core_class.py:
```diff
@@ -4,24 +4,28 @@
 Model
     Loads, reshapes, and holds SAMI2 output for a given model run
     specified by the user.
 -------------------------------------------------------------------------------
```
Recommend removing lower line and not extending upper line beyond header text.

aburrell commented on sami2py/_core_class.py:
```diff
@@ -93,26 +90,23 @@ def __init__(self, tag, lon, year, day, outn=False, test=False):
         out.append(('Day {day:03d}, {year:4d}').format(day=self.day,
                                                        year=self.year))
-        out.append(('Longitude = {lon:d} deg').format(lon=self.lon0))
+        out.append(('Longitude = {lon:5.1f} deg').format(lon=self.lon0))
```
```suggestion
        out.append(('Longitude = {:5.1f} deg').format(self.lon0))
```

Since you aren't really using the named functionality of format.

aburrell replied:
Also elsewhere.  But it's fine to keep it if you like this style.

aburrell commented on sami2py/_core_class.py:
```diff
@@ -164,7 +157,6 @@ def _calculate_slt(self):
     def _load_model(self):
         """Loads model results
-
         Returns
         -------
         void
```
Cut, here and elsewhere.

aburrell commented on sami2py/_core_class.py:
```diff
@@ -410,11 +439,9 @@ def plot_lat_alt(self, time_step=0, species=1):
 
         warnings.warn(' '.join(["Model.plot_lat_alt is deprecated and will be",
-                                "removed in a future version. ",
+                                "removed in version 0.3.0. ",
```
```suggestion
                                "removed in version 0.3.0+. ",
```

aburrell replied:
Just in case you bump up to 1.0.0 for some reason.

aburrell commented on sami2py/fortran/nrlmsise00_modified.f:
```diff
@@ -1183,7 +1183,8 @@ FUNCTION GLOBE7(YRD,SEC,LAT,LONG,TLOC,F107A,F107,AP,P)
       TINF=P(31)
       DO 50 I = 1,NSW
-   50 TINF = TINF + ABS(SW(I))*T(I)
+        TINF = TINF + ABS(SW(I))*T(I)
+   50   CONTINUE
```
Did you contact Emmert about this?

aburrell commented on sami2py/tests/test_core_class.py:
```diff
@@ -1,10 +1,13 @@
 import os
+
 import xarray as xr
-import sami2py
+import matplotlib
 import pytest
```
Not alphabetical

aburrell commented on sami2py/tests/test_utils.py:
```diff
@@ -127,3 +120,80 @@ def file_open_error(self):
+        del self.x, self.coeffs
+
+    def test_cos(self):
+        """Test the cos function when coeffs are all 0s except for one with 1
+        value
```
Recommend rewriting to span 1 line only, since second line won't appear when tests fail.

aburrell commented on sami2py/utils.py:
```diff
@@ -24,6 +27,9 @@
 """
 
 import os
+import warnings
+import numpy as np
+from scipy.optimize import curve_fit
```
Not alphabetical

aburrell commented on sami2py/utils.py:
```diff
@@ -113,6 +126,39 @@ def set_archive_dir(path=None, store=True):
+        """simple sine"""
+        return np.sin(n * np.pi * x / 12.0)
+
+    NumF = coeffs.shape
```
not snake case

aburrell commented on sami2py/utils.py:
```diff
@@ -150,3 +195,64 @@ def get_unformatted_data(dat_dir, var_name, reshape=False, dim=(0, 0)):
+
+def __make_fourier(na, nb):
+    """ The function for the curve fit
+    Parameters
```
```suggestion

    Parameters
```

aburrell replied:
Here and elsewhere

aburrell commented on sami2py/utils.py:
```diff
@@ -150,3 +195,64 @@ def get_unformatted_data(dat_dir, var_name, reshape=False, dim=(0, 0)):
+    """ The function for the curve fit
+    Parameters
+    ----------
+    na: (int)
+        number of cosine terms/coefficients
+    nb: (int)
```
```suggestion
    na : (int)
        number of cosine terms/coefficients
    nb : (int)
```

aburrell commented on setup.cfg:
```diff
@@ -21,8 +21,8 @@ classifiers =
   Programming Language :: Python :: 3.7
   Operating System :: OS Independent
```
Are you going to add Mac and more unix/linux system tests to back this up?

jklenzing replied:
Updated as linux / mac / windows.  The mac tests fail on github actions due to a configuration issue (#149).  Willing to keep the mac descriptor as I run locally on a mac.

aburrell commented on setup.cfg:
```diff
@@ -38,9 +38,10 @@ install_requires =
-  docs/conf.py ALL
+[flake8]
+max-line-length=80
+ignore =
+  W503
```
You could put this on one line

aburrell commented on setup.py:
```diff
@@ -38,12 +40,18 @@
+    with open(exb_path, 'w') as exb:
+        exb.writelines("%s\n" % line for line in zero_list)
+
 with open(os.path.join(file_path, 'fortran_path.txt'), 'w+') as f:
     f.write(fortran_path)
```
Recommend using a more descriptive file handle variable name.

aburrell commented on test_requirements.txt:
```diff
@@ -0,0 +1,10 @@
+coveralls
+flake8
+pytest
+pytest-cov
+pytest-ordering
+sphinx
+sphinx_rtd_theme
+numpydoc
+ipython
+m2r2
```
Not alphabetical

aburrell reviewed (commented):
Mostly style suggestions and questions.

aburrell commented:
> EDIT: fortran is required for the install, this is likely not to install clean unless you have this set up already. Maybe not even then. :/

What do you mean by "this"?

jklenzing commented on docs/requirements.txt:
```diff
@@ -1,2 +1,3 @@
 numpydoc
 ipython
+m2r2
```
```suggestion
ipython
m2r2
numpydoc
```

jklenzing commented:
> > EDIT: fortran is required for the install, this is likely not to install clean unless you have this set up already. Maybe not even then. :/
> 
> What do you mean by "this"?

having a fortran compiler set up

jklenzing commented:
Cleaned up the extra variables in init.  We have two parameters `archive_dir` and `archive_path`.  `archive_path` points to where the archive directory path (`archive_dir`) is stored.  This needs to be saved for some of the unit tests, so I changed it to `_archive_path` to avoid confusing users.  Could be cleaned up for the next version.
